### ISSUE TITLE
Allow properties to be defined as not saved on update by default

### ISSUE THREAD

**Issue by dantman** [sequelize/sequelize#3192]
Most database structures have have at least one column in one of their tables which by definition either **never** updates or is only ever updated in very explicit circumstances.

For example the `userid:STRING` of a User table nearly never changes, you'd only ever change it explicitly in a user rename call.

And if you have something like an Emails table, the `email:STRING` would by definition never change since it's a row describing that email and any email change would be a row removal/addition.

Running `.save()` on these models results in the `UPDATE` doing a set for these properties. This is unnecessary, most often unwelcome, and doesn't match what a developer would normally write.

One option would be to let define accept a column option that says not to save the column on update by default. Then an `UPDATE` on a normal `.save()` or `.save(['otherColumn'])` would omit updating that column. But the column could still be explicitly updated by including it in the array of columns to save.

Not sure what to name it.

**Comment by mickhansen** [comment]
What version are you using?
`.save()` should not be updating unchanged fields anymore by default.
In any case `save` does support a `fields` option that you could use.

**Comment by dantman** [comment]
I'm using `2.0.0-rc8` right now. I upgraded from `2.0.0-rc2` and `2.0.0-dev13`. And at one point in the past was using `1.7.9`.

I wasn't aware that Sequelize had updated to doing full delta saves.

I guess I should also re-examine the behavior of the methods that do implicit "set these properties, then save them". Since .setAttributes or whatever instances had back in `1.7.9` is gone now.

**Comment by mickhansen** [comment]
`setAttributes` was refactored to map to `set`

**Comment by dantman** [comment]
I think you mean `.update`.

I remember in the past there was a method, which I believe was `.setAttributes`, I avoided because while it's purpose was to implicitly accept an object of values to change and then implicitly save them, calling it would also save other unrelated attributes including the ones you don't want saved by default. So I started explicitly setting my properties and then running save with the list of fields to update.
